src/main/java/stsjorbsmod/actions/HealOnFatalAttackAction.java

```diff
@@ -0,0 +1,39 @@
+package stsjorbsmod.actions;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.HealAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.powers.MinionPower;
+import com.megacrit.cardcrawl.vfx.combat.FlashAtkImgEffect;
+
+public class HealOnFatalAttackAction extends AbstractGameAction {
+    private DamageInfo info;
+    private int heal;
+
+    public HealOnFatalAttackAction(AbstractCreature target, DamageInfo info, int healAmount) {
+        this.info = info;
+        this.setValues(target, info);
+        this.heal = healAmount;
+        this.actionType = ActionType.DAMAGE;
+        this.duration = 0.1F;
+
+    }
+
+    public void update() {
+        if (this.duration == 0.1F && this.target != null) {
```

Comment:
I think for the timing of the effects it might make more sense to be consistent with `VampireDamageAction` (ie, duration of `Settings.ACTION_DUR_MED`, do the `FlashAtkImgEffect` at the start of the duration, then `tickDuration()`, then if `isDone` execute the damage + heal effects